### PR TITLE
[coqdep] Cleanup findlib-specific functions.

### PR DESCRIPTION
**Pull request by ejgallego**
Code is a bit clearer now, and a step towards a more unified setup.

### CODE REVIEW

**Comment by SkySkimmer** [comment]
typo in commit message / PR title

File: tools/coqdep/lib/fl.ml [hunk 1]
```diff
@@ -36,27 +36,33 @@ module Error = struct
...
-    m
+    Some m
   with
+  (* This should not be necessary, but there's a problem in the platform build *)
+  | Sys_error _msg -> None
   (* findlib >= 1.9.3 uses its own Error exception, so we can't catch
      it without bumping our version requirements. TODO pass the message on once we bump. *)
   | _ -> Error.cannot_parse_meta_file package ""
 
+let rec find_META meta_files package =
```
Comment (gares) [inline tools/coqdep/lib/fl.ml:54]: maybe the correct name is the find_parseable_META

Comment (ejgallego) [reply tools/coqdep/lib/fl.ml:54]: Yes, amended. Something weird is going on in Windows, when we change this to find meta + parse the thing doesn't work. Seems very fishy and will likely be a blocker to remove the legacy layer.

Unfortunately I am at a loss on how to debug our Windows build, the platform job doesn't seem to store the logs @MSoegtropIMC  correct?

**Comment by ejgallego** [comment]
This is ready (CI failure due to missing 4.14 overlays)

**Review by Alizter — APPROVED** [review]

**Comment by Alizter** [comment]
I'm happy with merging this soon.

@gares Do you want to have another look?

**Comment by gares** [comment]
LGTM

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**